istioctl/pkg/util/handlers/handlers_test.go

```diff
@@ -56,6 +56,42 @@ func TestInferPodInfo(t *testing.T) {
 			wantPodName:   "istio-security-post-install-1.2.2-bm9w2",
 			wantNamespace: "istio-system",
 		},
+		{
+			proxyName:     "service/istiod",
+			namespace:     "",
+			wantPodName:   "service/istiod",
+			wantNamespace: "",
+		},
+		{
+			proxyName:     "service/istiod",
+			namespace:     "namespace",
+			wantPodName:   "service/istiod",
+			wantNamespace: "namespace",
+		},
+		{
+			proxyName:     "service/istiod.istio-system",
+			namespace:     "namespace",
+			wantPodName:   "service/istiod",
```

Comment:
I am confused about the pod names here you added

Reply:
I think the problem is before this function returned "pod name" but was later changed to "resource name" without updating the tests. I can fix it but it's not really in scope of my PR.

Reply:
Can you also explain what does below cmd do with this pr?

`istioctl pc listener gateway.gateway.networking.k8s.io/istiod`

Reply:
The same thing `svc/xyz`, `deployment/xyz`, etc. Gets a pod associated with that resource. There is a direct mapping from Service to Pod, and similarly there is a direct mapping from Gateway to Pod which we use here.

This allows users to think about higher level construct (Gateway) and not worry about implementation detail (pod name)

Reply:
IC, that is to say i can use istio gateway as well, though I do not think anyone will use it like this.